iijpta/ngx_http_secure_token_iijpta.c

```diff
@@ -190,18 +197,38 @@ ngx_secure_token_iijpta_get_var(
 		goto error;
 	}
 	outp += out_len;
-
 	out_len = outp - out;
-	p = ngx_pnalloc(r->pool, sizeof("pta=") + (out_len * 2));
-	if (p == NULL)
+
+	if (conf->avoid_cookies == 0)
 	{
-		goto error;
+	        p = ngx_pnalloc(r->pool, sizeof("pta=") + (out_len * 2) +
+				SEMICOLON_SPC + EXPIRES_SIZE +
+				SEMICOLON_SPC + MAX_AGE_SIZE);
+		if (p == NULL)
+		{
+		    goto error;
+		}
+
+		v->data = p;
+		p = ngx_copy(p, "pta=", sizeof("pta=") - 1);
+		p = ngx_hex_dump(p, out, out_len);
```

Comment:
Once you implement the previous comment, the code from ngx_pnalloc up to here can be shared by both cases (cookies/query)